### PR TITLE
Switch to negroni middleware

### PR DESCRIPTION
**Pull request by zdylag**
Using negroni middleware is nice since we can use it
independently of the router we choose. This will make it easier
to set a negroni middleware stack on a gorrilla.Mux.

### CODE REVIEW

File: middleware/mux_adapter/mux_adapter_test.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	"net/http/httptest"
+	"testing"
+
+	"github.com/gorilla/mux"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/stretchr/testify/assert"
+	"github.com/urfave/negroni"
+)
+
+func TestPrometheusMiddlewareAttached(t *testing.T) {
```
Comment (FrozenSake) [inline middleware/mux_adapter/mux_adapter_test.go:14]: I don't think this tests prometheus :)

File: middleware/mux_adapter/mux_adapter_test.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	router := mux.NewRouter()
+	router.Use(Middleware(negroni.HandlerFunc(testMiddleware)))
+	router.Path("/test").Handler(promhttp.Handler())
+
+	t.Log("Launching server")
+	ts := httptest.NewServer(router)
+	defer ts.Close()
+	metricsUrl := ts.URL + "/test"
+
+	_, err := http.Get(metricsUrl)
```
Comment (FrozenSake) [inline middleware/mux_adapter/mux_adapter_test.go:30]: should probably be testUrl

File: middleware/mux_adapter/mux_adapter_test.go (hunk 1 above)
Comment (zdylag) [inline middleware/mux_adapter/mux_adapter_test.go:14]: This should be renamed.

**Review by FrozenSake — APPROVED** [review]
Looks good for content, just a little nit: on the test names.

**Comment by zdylag** [comment]
@FrozenSake I renamed things!

**Review by FrozenSake — APPROVED** [review]
Nit: lingering comment, otherwise good.

**Comment by zdylag** [comment]
@FrozenSake and I synced offline: The lingering comment was that a random extra comment was on the end of a line. This is now removed I think.

**Review by Matt-Hurd — APPROVED** [review]

**Pull request merged**